devops/scripts/benchmarks/utils/flamegraph.py

```diff
@@ -0,0 +1,436 @@
+# Copyright (C) 2025 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import shutil
+import subprocess
+import signal
+import time
+
+from options import options
+from utils.utils import run, git_clone
+from utils.logger import log
+
+from datetime import datetime, timezone
+
+
+class FlameGraph:
+    """FlameGraph wrapper for managing FlameGraph tool execution and results."""
+
+    # FlameGraph SVG width to fit within web interface container
+    # Optimized width to avoid horizontal scrollbars within 1100px container
+    FLAMEGRAPH_WIDTH = 1000
+
+    def __init__(self):
+        self.timestamp = (
+            datetime.now(tz=timezone.utc).strftime(options.TIMESTAMP_FORMAT)
+            if options.timestamp_override is None
+            else options.timestamp_override
+        )
+
+        log.info("Downloading FlameGraph...")
+        repo_dir = git_clone(
+            options.workdir,
+            "flamegraph-repo",
+            "https://github.com/brendangregg/FlameGraph.git",
+            "master",
+        )
+
+        # FlameGraph doesn't need building, just verify scripts exist and are executable
```

Comment:
We don't check this anywhere else. I'm not sure if this would ever find an issue?